ouroboros-consensus/ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Protocol/Praos.hs

```diff
@@ -115,39 +116,85 @@ praosValidateView :: ( HasHeader    hdr
 praosValidateView getFields hdr =
     PraosValidateView (blockSlot hdr) (getFields hdr) (headerSigned hdr)
 
-forgePraosFields :: ( HasNodeState (Praos c) m
-                    , MonadRandom m
+{-------------------------------------------------------------------------------
+  Forging
+-------------------------------------------------------------------------------}
+
+data PraosNodeState c =
+    -- | The KES key is available
+    PraosKeyAvailable (SignKeyKES (PraosKES c))
+
+    -- | The KES key is being evolved
```

Comment:
Comment: any thread that modifies the state and sees this value should back off and wait.